build.sh

```diff
@@ -1,29 +1,29 @@
 set -e
 
-if [[ "$OSTYPE" == "darwin"* ]]; then
-	brew install md5sha1sum automake autogen doxygen
-elif [[ "$OSTYPE" == "linux"* ]]; then
-	if [[ "$GITHUB_ACTIONS" == "true" ]]; then
-		sudo apt-get install -y autopoint
-		sudo apt-get install -y doxygen
-	fi
-else
-	echo "Unknown OS type." >&2
-	exit 1
+if [[ "$OSTYPE" != "darwin"* && "$OSTYPE" != "linux"* ]]; then
+    echo "Error: Not supported OS type." >&2
+    exit 1
 fi
 
-git clone https://github.com/microsoft/vcpkg.git
-cd vcpkg
-git checkout tags/2023.11.20
-if [[ "$OSTYPE" == "darwin"* ]]; then
-	sed -i -e 's/0.63/1.3.0/' ports/vcpkg-tool-meson/vcpkg.json
-	sed -i -e 's/0.63.0/1.3.0/' ports/vcpkg-tool-meson/portfile.cmake
-	sed -i -e 's/bb91cea0d66d8d036063dedec1f194d663399cdf/7368795d13081d4928a9ba04d48498ca2442624b/' ports/vcpkg-tool-meson/portfile.cmake
-	sed -i -e 's/e5888eb35dd4ab5fc0a16143cfbb5a7849f6d705e211a80baf0a8b753e2cf877a4587860a79cad129ec5f3474c12a73558ffe66439b1633d80b8044eceaff2da/b2dc940a8859d6b0af8cb762c896d3188cadc1e65e3c7d922d6cb9a4ed7a1be88cd4d51a8aa140319a75e467816e714c409cf74c71c830bbc5f96bb81c1845ce/' ports/vcpkg-tool-meson/portfile.cmake
-	sed -i -e "s/intl_factory/packages['intl'] = intl_factory/" ports/vcpkg-tool-meson/meson-intl.patch
+if [[ "$DOWNLOAD_VCPKG" != "0" ]]; then
+	if [[ -d vcpkg ]]; then
+		echo "Error: vcpkg directory already exists. Remove it or set DOWNLOAD_VCPKG=0 if you are sure directory content is correct." >&2
+		exit 1
+	fi
+	git clone https://github.com/microsoft/vcpkg.git
+	cd vcpkg
+	git checkout tags/2024.01.12
+	if [[ "$OSTYPE" == "darwin"* ]]; then
+		sed -i -e 's/0.63/1.3.0/' ports/vcpkg-tool-meson/vcpkg.json
+		sed -i -e 's/0.63.0/1.3.0/' ports/vcpkg-tool-meson/portfile.cmake
+		sed -i -e 's/bb91cea0d66d8d036063dedec1f194d663399cdf/7368795d13081d4928a9ba04d48498ca2442624b/' ports/vcpkg-tool-meson/portfile.cmake
+		sed -i -e 's/e5888eb35dd4ab5fc0a16143cfbb5a7849f6d705e211a80baf0a8b753e2cf877a4587860a79cad129ec5f3474c12a73558ffe66439b1633d80b8044eceaff2da/b2dc940a8859d6b0af8cb762c896d3188cadc1e65e3c7d922d6cb9a4ed7a1be88cd4d51a8aa140319a75e467816e714c409cf74c71c830bbc5f96bb81c1845ce/' ports/vcpkg-tool-meson/portfile.cmake
+		sed -i -e 's/remove-freebsd-pcfile-specialization.patch/#/' ports/vcpkg-tool-meson/portfile.cmake
+		sed -i -e "s/intl_factory/packages['intl'] = intl_factory/" ports/vcpkg-tool-meson/meson-intl.patch
+	fi
+	./bootstrap-vcpkg.sh
+	cd ..
```

Comment:
This is for upgrading Meson to version 1.3.0 because 0.63 does not work with versions of python. It will be upgraded officially in newer versions of vcpkg. You can suggest required comments.

Reply:
Added in 05b02a9